lib/src/clojure_lsp/internal_api.clj

```diff
@@ -56,7 +56,10 @@
                            :type type
                            :extra extra}]
       (show-message-cli options message-content)))
-  (register-capability [_this _capability]))
+  (register-capability [_this _capability])
+
+  clojure-producer/IClojureProducer
+  (refresh-test-tree [_this _uris]))
```

Comment:
Couldn't we don't implement `IClojureProducer` at all here since we don't have test-tree on API? 
Or would that break something, since somewhere will try to call this function from a protocol that this record doesn't implement it?

Reply:
Yeah, the problem is that the server needs to implement the function and the lib needs to call it. And we do have it on the API, `ClojureLspProducer` in server.clj implements it. I'm also not happy with the way this part is working.

The good thing is that this is all purely Clojure, with no lsp4j or java in the way. So I think we can come up with something better here.
I do like the producer design, so maybe we should just have two of those. One is provided by lsp4clj with a default implementation for built-in LSP things and a second one for custom messages. The custom one can look however you want since the implementation and the calling side is fully outside of lsp4clj. lsp4clj would just handle construction and clean up for you.
WDYT @ericdallo?

Reply:
I've created a new issue to do this in the next version: #818